### PR TITLE
Add seq and Task.forEach

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
See [zulip discussion](https://roc.zulipchat.com/#narrow/stream/304902-show-and-tell/topic/article.20about.20looping.20in.20Roc/near/412638345)

### CODE REVIEW

File: platform/Task.roc [hunk 1]
```diff
@@ -208,3 +210,36 @@ batch = \current -> \next ->
...
+##
+## ```
+## authors : List ID
+## getAuthor : ID -> Task Author [DbError]
+##
+## getAuthors : Task (List Author) [DbError]
+## getAuthors = Task.list authors getAuthor
+## ```
+##
+list : List a, (a -> Task c d) -> Task (List c) d
```
Comment (rtfeldman) [inline platform/Task.roc:224]: What if we call this `Task.seq` (for "sequential") and have it just be `List (Task ok err) -> Task ok err`?

That way you can transform the list however you want (e.g. with `List.map`) first, and then `|> Task.seq` to run them all sequentially.

(We might also want a `Task.concurrent` or something in the future as well, once we can support that in the host.)

Comment (lukewilliamboswell) [reply platform/Task.roc:224]: What do we do then with the intermediate values between tasks? Should we throw them away and keep the final value? Or maybe restrict it to `Task {} err`'s so there is no value returned?

Comment (rtfeldman) [reply platform/Task.roc:224]: Oops, I guess that should be:

```elm
List (Task ok err) -> Task (List ok) err
```

Comment (lukewilliamboswell) [reply platform/Task.roc]: Updated to `Task.seq` ✅

**Review by rtfeldman — APPROVED** [review]
Thanks @lukewilliamboswell! 🎉

**Pull request merged**